### PR TITLE
Update build.xm for scriptdef

### PR DESCRIPTION
**Pull request by ashishappirio**
Added a single script tag that can be called by specific tests targets just by passing login details.

### CODE REVIEW

File: build.xml [hunk 1]
```diff
@@ -20,6 +20,57 @@
...
 -->
 
+  <scriptdef name="scepcificTest" language="javascript">
+    <attribute name="cmcuser"/>
+    <attribute name="cmcpass"/>
+    <attribute name="serverurl"/>
+    <![CDATA[
+    	//*** CREATE DEPLOY TASK ****
+		var deployTask = project.createTask('antlib:com.salesforce:deploy');
+		var username = attributes.get("cmcuser");
```
Comment (bruzzi) [inline build.xml]: Let's more generically call this attribute "username"

File: build.xml [hunk 2]
```diff
@@ -20,6 +20,57 @@
...
+		var password = attributes.get("cmcpass");
```
Comment (bruzzi) [inline build.xml]: Let's more generically call this attribute "password"

File: build.xml [hunk 3]
```diff
@@ -20,6 +20,57 @@
...
+		var version = project.getProperty("version");
+		var specifiedTest = project.getProperty("specifiedTests");
+		
+		if(username!=null && password!=null && serverurl!=null && source!=null && version!=null && specifiedTest!=null){
+			deployTask.setUsername(username);
+			deployTask.setPassword(password);
+    	    deployTask.setServerURL(serverurl);
+    	    deployTask.setDeployRoot(source + " sprint " + version);
+    	    deployTask.setTestLevel("RunSpecifiedTests");
+    	    deployTask.setMaxPoll("5000");
```
Comment (bruzzi) [inline build.xml]: Minor formatting issue, but looks like you have some spacing/tabbing issues between lines 37-44.

File: build.xml [hunk 4]
```diff
@@ -20,6 +20,57 @@
     </target>
 -->
 
+  <scriptdef name="scepcificTest" language="javascript">
```
Comment (bruzzi) [inline build.xml]: Typo in name: "scepcificTest". Probably best to name this script as a verb (e.g., "runSpecifiedTests").

File: build.xml [hunk 5]
```diff
@@ -20,6 +20,57 @@
     </target>
 -->
 
+  <scriptdef name="scepcificTest" language="javascript">
+    <attribute name="cmcuser"/>
+    <attribute name="cmcpass"/>
```
Comment (bruzzi) [inline build.xml]: Let's more generically call this attribute "password"

File: build.xml [hunk 6]
```diff
@@ -20,6 +20,57 @@
     </target>
 -->
 
+  <scriptdef name="scepcificTest" language="javascript">
+    <attribute name="cmcuser"/>
```
Comment (bruzzi) [inline build.xml]: Let's more generically call this attribute "username"

File: build.xml [hunk 7]
```diff
@@ -210,25 +323,25 @@
...
-      pollWaitMillis="20000"
-      maxPoll="5000"
-      checkOnly="${checkOnly}"
-      testLevel="RunSpecifiedTests"
-      deployRoot="${source} sprint ${version}">
-      <runTest>${specifiedTest}</runTest> <!--PresalesRequestHandler_Test-->
-    </sf:deploy>
-  </target>
+	<!-- deploy script -->
+	<scepcificTest cmcuser="${cmc.full.username}" cmcpass="${cmc.full.password}" serverurl="${sandbox.serverurl}"/>
```
Comment (bruzzi) [inline build.xml]: Typo "scepcificTest".  Update names of username/password attributes per earlier comments.

File: build.xml [hunk 8]
```diff
@@ -312,33 +425,23 @@
...
+      <antcall target="backup-prod">
+        <param name="version" value="${version}" />
+        <param name="type" value="destructiveChanges" />
+      </antcall>
+      <antcall target="copy-destructiveChanges">
+        <param name="version" value="${version}" />
+        <param name="source" value="${source}" />
+      </antcall>
+  	<!-- deploy script -->
+  	<scepcificTest cmcuser="${cmc.prod.username}" cmcpass="${cmc.prod.password}" serverurl="${prod.serverurl}"/>
```
Comment (bruzzi) [inline build.xml]: Typo "scepcificTest".  Update names of username/password attributes per earlier comments.

File: build.xml [hunk 9]
```diff
@@ -390,6 +514,25 @@
...
+	 <antcall target="backup-dev">
+       <param name="version" value="${version}" />
+       <param name="type" value="destructiveChanges" />
+     </antcall>
+     <antcall target="copy-destructiveChanges">
+       <param name="version" value="${version}" />
+       <param name="source" value="${source}" />
+     </antcall>
+     <!-- deploy script -->
+	 <scepcificTest cmcuser="${cmc.dev.username}" cmcpass="${cmc.dev.password}" serverurl="${sandbox.serverurl}"/>
```
Comment (bruzzi) [inline build.xml]: Typo "scepcificTest".  Update names of username/password attributes per earlier comments.

File: build.xml [hunk 10]
```diff
@@ -192,13 +264,54 @@
...
+      <antcall target="backup-qa">
+        <param name="version" value="${version}" />
+        <param name="type" value="destructiveChanges" />
+      </antcall>
+      <antcall target="copy-destructiveChanges">
+        <param name="version" value="${version}" />
+        <param name="source" value="${source}" />
+      </antcall>
+  	<!-- deploy script -->
+  	<scepcificTest cmcuser="${cmc.full.username}" cmcpass="${cmc.full.password}" serverurl="${sandbox.serverurl}"/>
```
Comment (bruzzi) [inline build.xml]: Typo "scepcificTest".  Update names of username/password attributes per earlier comments.

**Review by bruzzi — CHANGES_REQUESTED** [review]
Thanks for creating the function to run specified tests - much cleaner!  I had a few minor comments and requests before we finalize.  

PS: With this new ant functionality, I see opportunities to modularize this entire build.xml file in the future!

**Comment by ashishappirio** [comment]
Thanks Chris for your review on this, I've made the changes as suggested.

**Pull request merged**